// rs-theme/lib/views/header.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.HeaderCrud = exports.HeaderSimple = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouterNamedRoutes = require('react-router-named-routes');

var _AppBar = require('material-ui/AppBar');

var _AppBar2 = _interopRequireDefault(_AppBar);

var _MenuItem = require('material-ui/MenuItem');

var _MenuItem2 = _interopRequireDefault(_MenuItem);

var _DropDownMenu = require('material-ui/DropDownMenu');

var _DropDownMenu2 = _interopRequireDefault(_DropDownMenu);

var _RaisedButton = require('material-ui/RaisedButton');

var _RaisedButton2 = _interopRequireDefault(_RaisedButton);

var _Toolbar = require('material-ui/Toolbar');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HeaderSimple = exports.HeaderSimple = function HeaderSimple(props) {
    return _react2.default.createElement(_AppBar2.default, { title: props.title, showMenuIconButton: false });
};

var HeaderCrud = exports.HeaderCrud = _react2.default.createClass({
    displayName: 'HeaderCrud',
    updateClass: function updateClass() {
        this.forceUpdate();
    },
    render: function render() {
        var _this = this;

        var props = this.props;
        return _react2.default.createElement(
            _Toolbar.Toolbar,
            null,
            _react2.default.createElement(
                _Toolbar.ToolbarGroup,
                { firstChild: true },
                _react2.default.createElement(
                    _DropDownMenu2.default,
                    { value: props.selectedMenu(),
                        onChange: function onChange(ev, index, val) {
                            props.onMenuChange(val).then(function () {
                                _this.updateClass();
                            });
                        } },
                    props.menuItems.map(function (i, idx) {
                        return _react2.default.createElement(_MenuItem2.default, { key: idx, value: i.value, primaryText: i.text });
                    })
                )
            ),
            _react2.default.createElement(
                _Toolbar.ToolbarGroup,
                null,
                _react2.default.createElement(_Toolbar.ToolbarTitle, { text: props.title }),
                _react2.default.createElement(_Toolbar.ToolbarSeparator, null),
                _react2.default.createElement(_RaisedButton2.default, {
                    label: props.buttonText,
                    containerElement: _react2.default.createElement(_reactRouterNamedRoutes.Link, { to: props.buttonLink.route }),
                    primary: true,
                    linkButton: true })
            )
        );
    }
});
