// rs-theme/lib/views/message.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.MessageDefault = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reflux = require('reflux');

var _reflux2 = _interopRequireDefault(_reflux);

var _Snackbar = require('material-ui/Snackbar');

var _Snackbar2 = _interopRequireDefault(_Snackbar);

var _underscore = require('underscore');

var _underscore2 = _interopRequireDefault(_underscore);

var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');

var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);

var _getMuiTheme = require('material-ui/styles/getMuiTheme');

var _getMuiTheme2 = _interopRequireDefault(_getMuiTheme);

var _rsTodo = require('rs-todo');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var muiTheme = (0, _getMuiTheme2.default)();
var MessageDefault = exports.MessageDefault = _react2.default.createClass({
    displayName: 'MessageDefault',

    mixins: [_reflux2.default.ListenerMixin],
    getInitialState: function getInitialState() {
        return {
            open: false,
            message: ''
        };
    },
    componentDidMount: function componentDidMount() {
        var self = this;
        _underscore2.default.each(_rsTodo.TodoActions, function (obj, key) {
            var message = _rsTodo.TodoMessages['after'.concat(key.charAt(0).toUpperCase().concat(key.slice(1)))];
            if (typeof message !== 'undefined') {
                self.listenTo(_rsTodo.TodoActions[key].completed, function (data) {
                    if (data instanceof Object && typeof data.tokens !== 'undefined') {
                        _underscore2.default.each(data.tokens, function (t, k) {
                            message = message.replace(new RegExp(":".concat(k), "g"), t);
                        });
                    }
                    self.state.message = message;
                    self.state.open = true;
                    self.forceUpdate();
                });
            }
        });
    },
    render: function render() {
        return _react2.default.createElement(
            _MuiThemeProvider2.default,
            { muiTheme: muiTheme },
            _react2.default.createElement(_Snackbar2.default, _extends({}, this.state, {
                autoHideDuration: 4000
            }))
        );
    }
});
