// rs-theme/lib/views/message.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.RSMessageDefault = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reflux = require('reflux');

var _reflux2 = _interopRequireDefault(_reflux);

var _reactRouter = require('react-router');

var _Snackbar = require('material-ui/Snackbar');

var _Snackbar2 = _interopRequireDefault(_Snackbar);

var _underscore = require('underscore');

var _underscore2 = _interopRequireDefault(_underscore);

var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');

var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);

var _getMuiTheme = require('material-ui/styles/getMuiTheme');

var _getMuiTheme2 = _interopRequireDefault(_getMuiTheme);

var _rsTodo = require('rs-todo');

var _client = require('rs-helper/lib/client');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var muiTheme = (0, _getMuiTheme2.default)();

var RSMessageDefault = exports.RSMessageDefault = _react2.default.createClass({
    displayName: 'RSMessageDefault',

    mixins: [_reflux2.default.ListenerMixin],
    getInitialState: function getInitialState() {
        return {
            open: false,
            message: ''
        };
    },
    componentDidMount: function componentDidMount() {
        var self = this;
        (0, _client.bootstrapMessage)(self, _rsTodo.TodoActions, _rsTodo.TodoMessages, function (message) {
            self.state.message = message;
            self.state.open = true;
            self.forceUpdate();
        }, this.props.hook, this.props.childrenHook);
    },
    componentWillUnmount: function componentWillUnmount() {
        this.close();
    },
    close: function close() {
        this.state.open = false;
        this.state.message = '';
        this.forceUpdate();
    },
    render: function render() {
        return _react2.default.createElement(
            _MuiThemeProvider2.default,
            { muiTheme: muiTheme },
            _react2.default.createElement(_Snackbar2.default, _extends({}, this.state, {
                autoHideDuration: 4000,
                onRequestClose: this.close
            }))
        );
    }
});
