// rs-todo/src/stores/todo.js
import Reflux from 'reflux';
import axios from 'axios';
import io from 'socket.io-client';
import {resource, baseUrl} from 'rs-config';

let id = 1;

export const messages = {
    afterCreate: 'Successfully created new todo item entitled :title',
    afterDelete: 'Successfully deleted a todo item'
}

export const actions = Reflux.createActions({ //must be constant
    "create": {asyncResult: true},
    "upsert": {asyncResult: true},
    "delete": {asyncResult: true},
    "sync": {asyncResult: true}
});

export const store = Reflux.createStore({ //must be constant
    listenables: actions,
    init() {
        const socket = io.connect(baseUrl);
        let self = this;
        this.records = [];
        this.refresh();
        socket.on('localTodo:changed', () => {
            self.refresh();
        });
    },
    refresh() {
        let self = this;
        resource.get('/localTodos').then((res) => {
            self.records = res.data;
            actions.sync.completed(res);
            self.trigger();
        }).catch((err) => {
            actions.sync.failed(err);
        });
    },
    onCreate(title){
        resource.put('/localTodos', {title: title}).then((res) => {
            actions.create.completed({...res, tokens: {title: res.data.title}})
            ;
        }).catch((err) => {
            actions.create.failed(err);
        });
    },
    onUpsert(todo){
        actions.upsert.completed();
    },
    onDelete(id){
        resource.delete('/localTodos'.concat('/'.concat(id))).then((res) => {
            actions.delete.completed(res);
        }).catch((err) => {
            actions.delete.completed(err);
        });
    }
});
