// rs-theme/src/views/rs-todo/add.js
import React from 'react';
import Reflux from 'reflux';
import {TodoStore, TodoActions} from 'rs-todo';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';
import {withRouter}from 'react-router';

export const RSMuiTodoAdd = withRouter(React.createClass({
    mixins: [
        Reflux.ListenerMixin,
        Reflux.connect(TodoStore, 'updateClass'),
    ],

    updateClass() {
        this.forceUpdate();
    },

    addCompleted(){
        this.props.router.push('todo');
    },

    componentDidMount(){
        this.listenTo(TodoActions.create.completed, function () {
            console.log(arguments);
        });
    },

    render() {
        return (
            <div>
                <TextField name="title" ref="title"/>
                <RaisedButton label="Add Todo" onClick={ev => TodoActions.create(this.refs.title.input.value)}
                              primary={true}/>
            </div>
        )
    }
}));

export const RSMuiTodoAddConfigurator = {
    default: {
        title: 'Create New Todo'
    }
}
