// rs-theme/src/views/rs-todo/add.js
import React from 'react';
import Reflux from 'reflux';
import {TodoStore, TodoActions} from 'rs-todo';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';
import {withRouter}from 'react-router';
import {MessageDefault} from '../message';

export const RSMuiTodoAdd = withRouter(React.createClass({
    mixins: [
        Reflux.ListenerMixin
    ],

    componentDidMount(){
        const self = this;
        self.listenTo(TodoStore, 'updateClass');
        self.listenTo(TodoActions.create.completed, () => {
            self.props.router.push('todo');
        });
    },

    updateClass() {
        this.forceUpdate();
    },

    render() {
        return (
            <div>
                <TextField name="title" ref="title"/>
                <RaisedButton label="Add Todo" onClick={ev => TodoActions.create(this.refs.title.input.value)}
                              primary={true}/>
            </div>
        )
    }
}));

export const RSMuiTodoAddConfigurator = {
    default: {
        title: 'Create New Todo'
    }
}
