// rs-todo/lib/stores/todo.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.store = exports.actions = undefined;

var _reflux = require('reflux');

var _reflux2 = _interopRequireDefault(_reflux);

var _loopbackApp = require('loopback-app');

var _loopbackApp2 = _interopRequireDefault(_loopbackApp);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var id = 1;

var Todo = _loopbackApp2.default.models.LocalTodo;

var actions = exports.actions = _reflux2.default.createActions({ //must be constant
    "create": { asyncResult: true },
    "upsert": { asyncResult: true },
    "delete": { asyncResult: true }
});

var store = exports.store = _reflux2.default.createStore({ //must be constant
    listenables: actions,
    init: function init() {
        this.records = [];

        this.refresh();

        Todo.observe('after save', this.refresh);
        Todo.observe('after delete', this.refresh);
    },
    refresh: function refresh() {
        var self = this;
        Todo.find({}, function (err, res) {
            self.records = res;
            self.trigger();
        });
    },
    onCreate: function onCreate(title) {
        var todo = Todo.create({ title: title });
        actions.create.completed(todo);
    },
    onUpsert: function onUpsert(todo) {
        Todo.upsert(todo);
    },
    onDelete: function onDelete(id) {
        Todo.destroyById(id);
    }
});
