// rs-todo/lib/stores/todo-reflux.js
'use strict';

var _redux = require('redux');

var _reflux = require('reflux');

var _reflux2 = _interopRequireDefault(_reflux);

var _reduxThunk = require('redux-thunk');

var _reduxThunk2 = _interopRequireDefault(_reduxThunk);

var _immutable = require('immutable');

var _immutable2 = _interopRequireDefault(_immutable);

var _loopbackApp = require('loopback-app');

var _loopbackApp2 = _interopRequireDefault(_loopbackApp);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var id = 1;

var Todo = _loopbackApp2.default.models.LocalTodo;

var actions = _reflux2.default.createActions(["create", "get", "fetch", "update", "delete", "connect"]);

var store = _reflux2.default.createStore({
    listenables: actions,
    init: function init() {
        this.records = [];

        this.refresh();

        Todo.observe('after save', this.refresh);
        Todo.observe('after delete', this.refresh);
    },
    refresh: function refresh() {
        var self = this;
        Todo.find({}, function (err, results) {
            self.records = results;
            self.trigger();
        });
    },
    onCreate: function onCreate(title) {
        Todo.create({ title: title });
    }
});
module.exports = {
    actions: actions,
    store: store
};
