// rs-theme/lib/views/rs-todo/add.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.RSMuiTodoAddConfigurator = exports.RSMuiTodoAdd = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reflux = require('reflux');

var _reflux2 = _interopRequireDefault(_reflux);

var _rsTodo = require('rs-todo');

var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');

var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);

var _RaisedButton = require('material-ui/RaisedButton');

var _RaisedButton2 = _interopRequireDefault(_RaisedButton);

var _TextField = require('material-ui/TextField');

var _TextField2 = _interopRequireDefault(_TextField);

var _reactRouter = require('react-router');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var RSMuiTodoAdd = exports.RSMuiTodoAdd = (0, _reactRouter.withRouter)(_react2.default.createClass({
    displayName: 'RSMuiTodoAdd',

    mixins: [_reflux2.default.ListenerMixin, _reflux2.default.connect(_rsTodo.TodoStore, 'updateClass')],

    updateClass: function updateClass() {
        this.forceUpdate();
    },
    addCompleted: function addCompleted() {
        this.props.router.push('todo');
    },
    componentDidMount: function componentDidMount() {
        this.listenTo(_rsTodo.TodoActions.create.completed, function () {
            console.log(arguments);
        });
    },
    render: function render() {
        var _this = this;

        return _react2.default.createElement(
            'div',
            null,
            _react2.default.createElement(_TextField2.default, { name: 'title', ref: 'title' }),
            _react2.default.createElement(_RaisedButton2.default, { label: 'Add Todo', onClick: function onClick(ev) {
                    return _rsTodo.TodoActions.create(_this.refs.title.input.value);
                },
                primary: true })
        );
    }
}));

var RSMuiTodoAddConfigurator = exports.RSMuiTodoAddConfigurator = {
    default: {
        title: 'Create New Todo'
    }
};
