// rs-theme/lib/views/rs-todo/components.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.RSMuiTodoList = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _header = require('../header');

var _header2 = _interopRequireDefault(_header);

var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');

var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);

var _List = require('material-ui/List');

var _Checkbox = require('material-ui/Checkbox');

var _Checkbox2 = _interopRequireDefault(_Checkbox);

var _FloatingActionButton = require('material-ui/FloatingActionButton');

var _FloatingActionButton2 = _interopRequireDefault(_FloatingActionButton);

var _add = require('material-ui/svg-icons/content/add');

var _add2 = _interopRequireDefault(_add);

var _deleteForever = require('material-ui/svg-icons/action/delete-forever');

var _deleteForever2 = _interopRequireDefault(_deleteForever);

var _colors = require('material-ui/styles/colors');

var _IconButton = require('material-ui/IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

var _moreVert = require('material-ui/svg-icons/navigation/more-vert');

var _moreVert2 = _interopRequireDefault(_moreVert);

var _IconMenu = require('material-ui/IconMenu');

var _IconMenu2 = _interopRequireDefault(_IconMenu);

var _MenuItem = require('material-ui/MenuItem');

var _MenuItem2 = _interopRequireDefault(_MenuItem);

var _rsTodo = require('rs-todo');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var RSMuiTodoList = exports.RSMuiTodoList = _react2.default.createClass({
    displayName: 'RSMuiTodoList',

    mixins: [Reflux.listenTo(_rsTodo.TodoStore, 'updateClass')],

    updateClass: function updateClass() {
        this.forceUpdate();
    },
    render: function render() {
        return _react2.default.createElement(
            _MuiThemeProvider2.default,
            { muiTheme: this.props.muiTheme },
            _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(_header2.default, { title: 'Todo List' }),
                _react2.default.createElement(
                    _List.List,
                    null,
                    _rsTodo.TodoStore.records.map(function (t) {
                        //every single function definition in es6 must be wrapped by () first on its argument(s)
                        var listStyle = t.completed ? { color: '#009688' } : {};
                        return _react2.default.createElement(_List.ListItem, {
                            key: t.id,
                            leftCheckbox: _react2.default.createElement(_Checkbox2.default, { defaultChecked: t.completed, onCheck: function onCheck(ev) {
                                    return _rsTodo.TodoActions.upsert(_extends({}, t.__data, { completed: ev.target.checked }));
                                } }),
                            primaryText: t.title,
                            style: listStyle,
                            rightIconButton: _react2.default.createElement(
                                _IconMenu2.default,
                                { iconButtonElement: _react2.default.createElement(
                                        _IconButton2.default,
                                        {
                                            touch: true,
                                            tooltip: 'more',
                                            tooltipPosition: 'bottom-center'
                                        },
                                        _react2.default.createElement(_moreVert2.default, { color: _colors.grey400 })
                                    ) },
                                _react2.default.createElement(
                                    _MenuItem2.default,
                                    { onClick: function onClick(ev) {
                                            return _rsTodo.TodoActions.delete(t.id);
                                        } },
                                    'Delete'
                                )
                            )
                        });
                    })
                ),
                _react2.default.createElement(
                    _FloatingActionButton2.default,
                    { style: { position: 'absolute', right: 20, bottom: 20 } },
                    _react2.default.createElement(_add2.default, null)
                )
            )
        );
    }
});
