// rs-theme/src/views/message.js
import React from 'react';
import Reflux from 'reflux';
import {withRouter} from 'react-router';
import Snackbar from 'material-ui/Snackbar';
import _ from 'underscore';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import {TodoActions, TodoMessages} from 'rs-todo';
import {bootstrapMessage} from 'rs-helper/lib/client';

const muiTheme = getMuiTheme();

export const RSMessageDefault = React.createClass({
    mixins: [
        Reflux.ListenerMixin
    ],
    getInitialState() {
        return {
            open: false,
            message: ''
        }
    },
    componentDidMount(){
        const self = this;
        bootstrapMessage(self, TodoActions, TodoMessages, (message) => {
            self.state.message = message;
            self.state.open = true;
            self.forceUpdate();
        }, this.props.hook, this.props.childrenHook);
    },
    componentWillUnmount(){
        this.close();
    },
    close() {
        this.state.open = false;
        this.state.message = '';
        this.forceUpdate();
    },
    render(){
        return (
            <MuiThemeProvider muiTheme={muiTheme}>
                <Snackbar
                    {...this.state}
                    autoHideDuration={4000}
                    onRequestClose={this.close}
                />
            </MuiThemeProvider>
        )
    }
});
